File: app/src/main/java/khorunzhyicom/alex/theposter/di/components/injector/CommandInjector.java
package khorunzhyicom.alex.theposter.di.components.injector;

import khorunzhyicom.alex.theposter.service.commands.GetNowPlayingMoviesCommand;
import khorunzhyicom.alex.theposter.service.commands.GetPopularMoviesCommand;
import khorunzhyicom.alex.theposter.service.commands.GetTopMoviesCommand;
import khorunzhyicom.alex.theposter.service.commands.GetUpcomingMoviesCommand;

public interface CommandInjector extends Injector {

    void inject(GetPopularMoviesCommand command);
    void inject(GetTopMoviesCommand command);
    void inject(GetNowPlayingMoviesCommand command);
    void inject(GetUpcomingMoviesCommand command);
}

File: app/src/main/java/khorunzhyicom/alex/theposter/di/components/injector/CommandInjector.kt
package khorunzhyicom.alex.theposter.di.components.injector

import khorunzhyicom.alex.theposter.service.commands.configuration.GetConfigurationCommand
import khorunzhyicom.alex.theposter.service.commands.movies.GetNowPlayingMoviesCommand
import khorunzhyicom.alex.theposter.service.commands.movies.GetPopularMoviesCommand
import khorunzhyicom.alex.theposter.service.commands.movies.GetTopMoviesCommand
import khorunzhyicom.alex.theposter.service.commands.movies.GetUpcomingMoviesCommand

interface CommandInjector : Injector {

    fun inject(command: GetPopularMoviesCommand)
    fun inject(command: GetTopMoviesCommand)
    fun inject(command: GetNowPlayingMoviesCommand)
    fun inject(command: GetUpcomingMoviesCommand)
    fun inject(command: GetConfigurationCommand)
}
